doc/source/whatsnew/v0.24.0.txt

```diff
@@ -487,6 +487,71 @@ Previous Behavior:
         0
     0 NaT
 
+.. _whatsnew_0240.api.dataframe_cmp_broadcasting:
+
+DataFrame Comparison Operations Broadcasting Changes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Previously, the broadcasting behavior of :class:`DataFrame` comparison
+operations (``==``, ``!=``, ...) was inconsistent with the behavior of
+arithmetic operations (``+``, ``-``, ...).  The behavior of the comparison
+operations has been changed to match the arithmetic operations in these cases.
+(:issue:`22880`)
+
+The affected cases are: operating against a 2-dimensional ``np.ndarray`` with
+either 1 row or 1 column, a list or tuple with the same length matching the
+number of rows in the :class:`DataFrame`, and a list or tuple with the
+length matching the number of columns in the :class:`DataFrame`.
+
+Previous Behavior:
+
+.. code-block:: ipython
+
+   In [3]: arr = np.arange(6).reshape(3, 2)
+   In [4]: df = pd.DataFrame(arr)
+
+   In [5]: df == arr[[0], :]  # comparison used to broadcast where arithmetic would raise
+   Out[5]:
+          0      1
+   0   True   True
+   1  False  False
+   2  False  False
+   In [6]: df + arr[[0], :]
+   ...
+   ValueError: Unable to coerce to DataFrame, shape must be (3, 2): given (1, 2)
+
+   In [7]: df == (1, 2)  # length matches number of columns; comparison used to raise where arithmetic would broadcast
```

Comment:
These code blocks are put in `<pre>` divs in the docs, so long lines will force horizontal scrolling / look extra bad on mobile. Better to continue the input with a

```
In [7]:  ...
   ...: # comment ...
   ...: # maybe multiline
```